sdk/identity/identity/src/msal/utils.ts

```diff
@@ -137,99 +142,74 @@ export function getMSALLogLevel(logLevel: AzureLogLevel | undefined): msalCommon
+        logger.info(formatError(scopes, error.message));
+        return new CredentialUnavailableError(error.message);
+      case "device_code_polling_cancelled":
+        return new AbortError("The authentication has been aborted by the caller.");
+      case "consent_required":
+      case "interaction_required":
+      case "login_required":
+        logger.info(
+          formatError(scopes, `Authentication returned errorCode ${msalError.errorCode}`),
+        );
+        break;
+      default:
+        logger.info(formatError(scopes, `Failed to acquire token: ${error.message}`));
+        break;
     }
-    ensureValidMsalToken(scopes, this.logger, result, getTokenOptions);
-    this.logger.getToken.info(formatSuccess(scopes));
-    return {
-      token: result!.accessToken!,
-      expiresOnTimestamp: result!.expiresOn!.getTime(),
-    };
   }
-
-  /**
-   * Handles MSAL errors.
-   */
-  protected handleError(scopes: string[], error: Error, getTokenOptions?: GetTokenOptions): Error {
-    if (
-      error.name === "AuthError" ||
-      error.name === "ClientAuthError" ||
-      error.name === "BrowserAuthError"
-    ) {
-      const msalError = error as msalCommon.AuthError;
-      switch (msalError.errorCode) {
-        case "endpoints_resolution_error":
-          this.logger.info(formatError(scopes, error.message));
-          return new CredentialUnavailableError(error.message);
-        case "device_code_polling_cancelled":
-          return new AbortError("The authentication has been aborted by the caller.");
-        case "consent_required":
-        case "interaction_required":
-        case "login_required":
-          this.logger.info(
-            formatError(scopes, `Authentication returned errorCode ${msalError.errorCode}`),
-          );
-          break;
-        default:
-          this.logger.info(formatError(scopes, `Failed to acquire token: ${error.message}`));
-          break;
-      }
-    }
-    if (
-      error.name === "ClientConfigurationError" ||
-      error.name === "BrowserConfigurationAuthError" ||
-      error.name === "AbortError"
-    ) {
-      return error;
-    }
-    if (error.name === "NativeAuthError") {
-      this.logger.info(
-        formatError(
-          scopes,
-          `Error from the native broker: ${error.message} with status code: ${
-            (error as any).statusCode
-          }`,
-        ),
-      );
-      return error;
-    }
-    return new AuthenticationRequiredError({ scopes, getTokenOptions, message: error.message });
+  if (
+    error.name === "ClientConfigurationError" ||
+    error.name === "BrowserConfigurationAuthError" ||
+    error.name === "AbortError"
+  ) {
+    return error;
+  }
+  if (error.name === "NativeAuthError") {
+    logger.info(
+      formatError(
+        scopes,
+        `Error from the native broker: ${error.message} with status code: ${
+          (error as any).statusCode
+        }`,
+      ),
+    );
+    return error;
   }
```

Comment:
Should we add the switch case of msal-error checks back? Because not for all cases we throw `AuthenticationRequiredError` and in some cases we want to log the msal error code as well.
```ts
switch (msalError.errorCode) {
        case "endpoints_resolution_error":
          this.logger.info(formatError(scopes, error.message));
          return new CredentialUnavailableError(error.message);
        case "device_code_polling_cancelled":
          return new AbortError("The authentication has been aborted by the caller.");
        case "consent_required":
        case "interaction_required":
        case "login_required":
          this.logger.info(
            formatError(scopes, `Authentication returned errorCode ${msalError.errorCode}`),
          );
          break;
        default:
          this.logger.info(formatError(scopes, `Failed to acquire token: ${error.message}`));
          break;
      }
```

Reply:
Sorry ignore my comment, just realized the msal switch case is there.